### PR TITLE
Add multi-theme build artifacts to Polaris tokens

### LINKED ISSUE #10043: Adjust Polaris tokens build artifacts (ESM, CJS, JSON, SCSS, CSS, etc.) to support theming
Polaris tokens currently exports multiple build targets (e.g. ESM, CJS, JSON, SCSS, and CSS)

- Investigate and gather (rough) usage metrics for each build target
- Determine if any targets can be deprecated
- Add theme support to each build target
- Outline a rollout strategy for any additions and deprecations

### PR DESCRIPTION
**Pull request by aaronccasanova**
Closes https://github.com/Shopify/polaris/issues/10043

- Updated `ESM` and `CJS` build targets to support theming
  * Exposed theme objects from a new `@shopify/polaris-tokens/themes` entry point (configured in the package.json)
  * Omitted `JSON` build targets as there are no usages of the existing exports (@samrose3 and I propose deprecating the entire build target in the next major release)
- Removed `valueExperimental` type and tokens as the overrides are now accessible from the `Polaris-Summer-Editions-2023` theme (see below)

#### Migrating from `metadata.<token-group>.<token-name>.valueExperimental`

**Before:**
```ts
import {metadata} from '@shopify/polaris-tokens';

metadata.color['color-bg'].valueExperimental
```

**After:**
```ts
import * as themes from '@shopify/polaris-tokens/themes';

themes.polarisSummerEditions2023.color['color-bg']
```

**Example ESM artifact**

Screenshot 2023-08-21 at 5 23 00 PM

### CODE REVIEW

File: polaris-tokens/package.json [hunk 1]
```diff
@@ -14,10 +14,22 @@
...
     "./css/*": "./dist/css/*",
     "./scss/*": "./dist/scss/*"
   },
+  "typesVersions": {
+    "*": {
+      "themes": [
+        "./dist/types/build/themes.d.ts"
+      ]
+    }
+  },
```
Comment (lgriffee) [inline polaris-tokens/package.json:32]: Added typesVersions to resolve themes types. 

**Before**
Screenshot 2023-08-22 at 10 54 12 AM

**After** 
Screenshot 2023-08-22 at 11 00 00 AM

**Review by sam-b-rose — APPROVED** [review]
💯 ✨

**Comment by aaronccasanova** [comment]
Closed in favor of https://github.com/Shopify/polaris/pull/10250

**Pull request closed without merging**